stubs/openpyxl/openpyxl/reader/excel.pyi

```diff
@@ -19,8 +25,14 @@ class ExcelReader:
     def read_workbook(self) -> None: ...
     def read_properties(self) -> None: ...
     def read_theme(self) -> None: ...
-    def read_chartsheet(self, sheet, rel) -> None: ...
+    def read_chartsheet(self, sheet: Chartsheet, rel: Relationship) -> None: ...
     def read_worksheets(self) -> None: ...
     def read(self) -> None: ...
 
-def load_workbook(filename, read_only: bool = ..., keep_vba=..., data_only: bool = ..., keep_links: bool = ...): ...
+def load_workbook(
+    filename: SupportsRead[bytes] | str,
```

Comment:
As pointed out by the `mypy_primer` diff, it looks like it's also okay to pass in `os.PathLike[str]` objects for the `filename` parameter, since `os.PathLike[str]` objects are accepted by `os.path.splitext`:

```pycon
>>> from pathlib import Path
>>> import os.path
>>> os.path.splitext(Path('foo/bar/baz.pyi'))[-1].lower()
'.pyi'
```

(Argument validation and type coercion for objects passed into `load_workbook` is done by the `_validate_archive` function here, which passes the `filename` argument to `os.path.splitext()` if it doesn't have a `read()` method: https://foss.heptapod.net/openpyxl/openpyxl/-/blob/branch/3.0/openpyxl/reader/excel.py#L67-97.)

We have a `StrPath` alias in the `_typeshed` directory that we use to describe the `str | PathLike[str]` union -- I recommend that you use it here. (You'll have to add `from _typeshed import StrPath` at the top of the file.)

```suggestion
    filename: SupportsRead[bytes] | StrPath,
```

Reply:
This, and `ExcelReader.__init__` is ultimately passed to `ZipFile` in `_validate_archive`.
`ZipFile.__init__` is typed as `ZipFile(file: StrPath | IO[bytes], ...)`

After double-checking, ZipFile could use some overloads. It only needs a name in "read" mode.